### PR TITLE
Polish StringOps module

### PR DESCRIPTION
**Pull request by asger-semmle**
- Adds the `::Range` pattern to the classes in `StringOps`.
- Fixes a bug in an `EndsWith` class
- Adds `StringOps::Concatenation` as a forwarder to `StringConcatenation` for now.

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/StringOps.qll [hunk 1]
```diff
@@ -5,339 +5,488 @@
 import javascript
 
 module StringOps {
-  /**
-   * A expression that is equivalent to `A.startsWith(B)` or `!A.startsWith(B)`.
-   */
-  abstract class StartsWith extends DataFlow::Node {
+  class StartsWith extends DataFlow::Node {
```
Comment (xiemaisi) [inline javascript/ql/src/semmle/javascript/StringOps.qll]: qldoc

File: javascript/ql/test/library-tests/StringConcatenation/ClassContainsTwo.ql [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import javascript
```
Comment (xiemaisi) [inline javascript/ql/test/library-tests/StringConcatenation/ClassContainsTwo.ql]: Is this test missing a `.expected` file?

**Comment by asger-semmle** [comment]
* Is the plan to deprecate/privatize `StringConcatenation::` later?

I'd like for `StringOps::Concatenation` to be main API for cases where this makes sense, but I don't think it will cover every case. Some predicates that deal directly with concatenation *operands*, like `getPreviousLeaf`, can't be converted to member predicates on `StringOps::Concatenation`.

* If so, why not introduce the `Range` pattern immediately here?

This is exactly what caused performance issues the first time I tried the `XXX::Range` pattern (called `CustomXXX` back then), and there's not enough time to find the root cause of that before feature freeze. But I would like a forward-compatible API to make it into this release so we can use it in cookbook queries.

**Comment by asger-semmle** [comment]
[Evaluation](https://git.semmle.com/asger/dist-compare-reports/tree/florian.ti.semmle.com_1551222076622) looks good.

**Pull request merged**